csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,31 @@
+
+def _csharp_resx_impl(ctx):
+    print("todo")
+
+csharp_resx = rule(
+    implementation = _csharp_resx_impl,
+    attrs = {
+        "src": attr.label(

```

Comment:
It might be more ergonomic for this to be `srcs` and to invoke `dotnet` multiple times in a loop behind the scenes.

Reply:
If we went this route, the API for specifying the `identifier` would need to be a [label_keyed_string_dict](https://docs.bazel.build/versions/master/skylark/lib/attr.html#label_keyed_string_dict), which makes the specification of the `identifier` mandatory.

I would also have concerns with the resource being used incorrectly with localized resx files. For localized resx files, satellite assemblies are created in directories based on the culture code. Those will need some special handling, as the pathing of those assemblies matter.